lib/runner.js

```diff
@@ -795,7 +795,7 @@ Runner.prototype.run = function (fn) {
   var rootSuite = this.suite;
 
   // If there is an `only` filter
-  if (this.hasOnly) {
+  if (hasOnly(rootSuite)) {
```

Comment:
Okay, so it looks like with the addition of `--forbid-only`, this may need some changes.

`Runner#hasOnly` is set if e.g. a `describe.only()` is encountered (see `lib/interfaces/common.js`).

The `--forbid-only` functionality will check this value.  At the point that it's checked, the `Runner` may or may not have a `suite` property.  This means there's likely some weird interactions between `--forbid-only` and `--watch`.

A solution may be to set `Runner#hasOnly` to `true` if the result of `hasOnly()` is `true`.  `hasOnly()` may also want to *check* the value of `Runner#hasOnly`.  When the `Runner`'s `end` event is handled, it may also need to reset `Runner#hasOnly` to `false`.

I haven't fiddled with this too much, but this is the matrix we're trying to assert is OK:

1.  `describe.only()`
2.  `it.only()`
3.  `--watch`
4.  `--forbid-only`

For example, run a `--watch` with `--forbid-only`, then add a `describe.only()`, save, then remove it again & save.  What happens?

Reply:
This should be merged right now, I think.